ddmrp/models/stock_buffer.py

```diff
@@ -702,11 +702,19 @@ def _compute_dlt(self):
             elif rec.buffer_profile_id.item_type == "distributed":
                 rec.dlt = rec.lead_days
             else:
-                rec.dlt = (
-                    rec.product_id.seller_ids
-                    and rec.product_id.seller_ids[0].delay
-                    or rec.lead_days
+                sellers = rec.product_id.seller_ids
+                sellers_variant = sellers.filtered(
+                    lambda s: s.product_id == rec.product_id
                 )
+                sellers_no_variant = sellers.filtered(lambda s: not s.product_id)
+                if sellers_variant:
+                    rec.dlt = sellers_variant[0].delay
+                elif sellers_no_variant:
+                    rec.dlt = sellers_no_variant[0].delay
+                elif sellers:
+                    rec.dlt = sellers[0].delay
+                else:
+                    rec.dlt = rec.lead_days
```

Comment:
```suggestion
                rec.dlt = rec._get_dlt_from_vendor()

    def _get_dlt_from_vendor(self):
        all_sellers = rec.product_id.seller_ids
        # specific for variant
        sellers = all_sellers.filtered(
            lambda s: s.product_id == rec.product_id
        )
        if not sellers:
            # generic no variant
            sellers = all_sellers.filtered(lambda s: not s.product_id)
        if not sellers:
            # fallback to all sellers
            sellers = all_sellers
        
        if sellers:
            return fields.first(sellers).delay
        return self.lead_days
```

Reply:
Don't you think your code is harder to read? My code didn't need any comments as it was self explanatory. Maybe I'm missing something else? I do agree that a dedicated method is always good though

Waiting your thoughts, thanks for the review :)

Reply:
The solution from @simahawk is better performance wise, as it will filter on some record only if it is necessary.

Reply:
exactly this :)

Reply:
in any case the code is quite easy to read even if you don't have specific var names.
What is missing is a good docstring instead. Would you mind adding it?

Reply:
Actually I realized that it is better to have that method to get the default supplier info record and the use it to compute both the main supplier and the dlt.

Would you mind reviewing again?